src/test/java/com/ericsson/gerrit/plugins/eiffel/handlers/DataBaseHandlerTest.java

```diff
@@ -0,0 +1,217 @@
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testInsertIntoExistingFails() throws Exception {
+        String firstEiffelEventId = generateEiffelEventId();
+        String secondEiffelEventId = generateEiffelEventId();
+        dbHandler.insertInto(Table.SCS_TABLE, branch, firstEiffelEventId);
+
+        try {
+            dbHandler.insertInto(Table.SCS_TABLE, branch, secondEiffelEventId);
+            assertEquals("Expected the call to throw and Exception but none was thrown!", true, false);
+        } catch (SQLException e) {
+            // test passed!
+        }
+
+    }
+
+    /**
+     * Empty db should return empty string
+     *
+     * @throws Exception
+     */
+    @Test(expected = NoSuchElementException.class)
+    public void testGetNoneExistingEventIdReturnsNull() throws Exception {
+        dbHandler.getEventID(Table.SCS_TABLE, branch);
+    }
+
+    /**
+     * Test Table enum
+     */
+    @Test
+    public void testTables() {
+        Table scsTable = Table.SCS_TABLE;
+        Table sccTable = Table.SCC_TABLE;
+
+        assertEquals("Table.SCS_TABLE tabler key should be", scsTableKey, scsTable.getKeyName());
+        assertEquals("Table.SCC_TABLE tabler key should be", sccTableKey, sccTable.getKeyName());
+
+    }
+
+    /**
+     * Test throwing several exceptions and ensure they cause the class to return
+     * correct values and or the exception is correctly caught. Exceptions caused by
+     * mocks should not leak into the test.
+     *
+     * @throws Exception
+     */
+    @Test(expected = NoSuchElementException.class)
+    public void testExceptionsIsThrown() throws Exception {
+        // Prepare mocks
+        Connection connection = mock(Connection.class);
+        PreparedStatement preparedStatement = mock(PreparedStatement.class);
+        Statement statement = mock(Statement.class);
+        ResultSet result = mock(ResultSet.class);
+
+
+        PowerMockito.mockStatic(DriverManager.class);
+        BDDMockito.given(DriverManager.getConnection(Mockito.any())).willReturn(connection);
+
+        if (result.next()) {
+            Mockito.doReturn(preparedStatement).when(connection).prepareStatement(Mockito.any());
+            Mockito.doReturn(result).when(preparedStatement).executeQuery();
+        } else {
+            throw new NoSuchElementException("Mocking ResultSet class was unsuccessful...!");
+        }
+
+        // When mocking an exception while executing getEventID the function should
+        // return an empty String,
+        // exception may be thrown when no values was found and should be empty.
+        if (result.next()) {
+            Mockito.when(result.next()).thenThrow(new SQLException("Exception thrown by test"));
+            exception.expect(RuntimeException.class);
+            dbHandler.getEventID(Table.SCS_TABLE, branch);
+        } else {
+            throw new NoSuchElementException("Mocking ResultSet class was unsuccessful...!");
+        }
+
```

Comment:
:eyes: Since the test throws an exception at this point, and test except exceptions here, test will pass but code below will never be executed. To make it more clear please split the test into several tests so it is easier to see that each test passes.

Reply:
It feels weird to even do these if/else statements. If the mocking fails, the test should fail? Is it okay if i don't break these old tests up now?

Reply:
Your call but if not now it will never happen.

Reply:
If/else were just meant to satisfy codacy. The problem was "else" which throw the exception. Now I try to fix it by mocking result.next(). You may come up with some other solutions.

Reply:
The problem here is that not all parts of the test is run since the NoSuchElementException is thrown before the below tests. This test needs to be split intoo several tests and test every exception on it´s own.

Reply:
next() is already mocked. We need to do this, otherwise codacy complains about the resultset.